src/gha_runner/__main__.py

```diff
@@ -1,13 +1,107 @@
 from gha_runner.clouddeployment import CloudDeploymentFactory
 from gha_runner.gh import GitHubInstance
-import sys
 import os
+import json
 
-# https://github.com/hukkin/tomli?tab=readme-ov-file#building-a-tomlitomllib-compatibility-layer
-if sys.version_info >= (3, 11):
-    import tomllib
-else:
-    import tomli as tomllib
+
+def env_parse_helper(
+    params: dict, var: str, key: str, is_json: bool = False
+) -> dict:
+    val = os.environ.get(var)
+    if val is not None:
+        if val != "":
+            if is_json:
+                params[key] = json.loads(val)
+            else:
+                params[key] = val
+    return params
```

Comment:
1. Checking that the logic is correct here. Is the intent, in fact, to not set `param[key]` at all if it is `None` or the empty string?
2. Simplify a bit by combining the first two `if` statements with an `and`.

Reply:
The goal of this function is to do nothing if the value is `None` or if it is an empty string. So you are correct that we can simplify this here.